### DISCUSSION TITLE
Build docker image from custom theia code

### DISCUSSION THREAD

**Discussion by youssef-chtourou** [eclipse-theia/theia#8597] [Q&A] [answered]
Hello Community

I've change some file in Theia source code to disable some function

Now , i would generate a Docker image from this code , can someone help me or someone hava a idea about it ??

Thanks a lot

**Comment by vince-fugnitto** [comment, marked as answer]
@chtourou-youssef it is not advised to directly modify the Theia source code and attempt to maintain it (including uplifting).
Instead, you should follow the guide on how to compose your own applications:
-  https://theia-ide.org/docs/composing_applications/

For the case of `docker`, you may refer to the number of examples we have under `theia-apps`: 
- https://github.com/theia-ide/theia-apps
